src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,157 @@
+                                                                           IDEDiagnosticIds.PreferBuiltInOrFrameworkTypeDiagnosticId,
+                                                                           IDEDiagnosticIds.AddBracesDiagnosticId,
+                                                                           IDEDiagnosticIds.AddAccessibilityModifiersDiagnosticId,
+                                                                           IDEDiagnosticIds.OrderModifiersDiagnosticId,
+                                                                           IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId,
+                                                                           IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForConstructorsDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForMethodsDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForConversionOperatorsDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForOperatorsDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                                                                           IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId,
+                                                                           IDEDiagnosticIds.InlineDeclarationDiagnosticId,
+                                                                           CSharpRemoveUnusedVariableCodeFixProvider.CS0168,
+                                                                           CSharpRemoveUnusedVariableCodeFixProvider.CS0219,
+                                                                           IDEDiagnosticIds.UseObjectInitializerDiagnosticId,
+                                                                           IDEDiagnosticIds.UseCollectionInitializerDiagnosticId);
+        
+        public CSharpCodeCleanupFixer(ICodeFixService codeFixService, IThreadingContext threadingContext, bool assertIsForeground = false)
+            : base(threadingContext, assertIsForeground)
+        {
+            _codeFixServiceOpt = codeFixService;
+        }
+
+        public async Task<bool> FixAsync(ICodeCleanUpScope scope, FixIdContainer enabledFixIds, CancellationToken cancellationToken)
+        {
+            var textBufferScope = scope as TextBufferCodeCleanUpScope;
+            if (textBufferScope == null)
+            {
+                var hierarchyContent = scope as IVsHierarchyCodeCleanupScope;
```

Comment:
this feels super weird.  why do we need to manually be downcasting these types?

Reply:
you can use a typeswitch to make this much cleaner.

Reply:
> why do we need to manually be downcasting these types?

This is a limitation of the current editor API

> you can use a typeswitch to make this much cleaner.

This is now used 👍